### PR TITLE
core: automatically detect linked detectors

### PR DESCRIPTION
**Pull request by multun**
Preparatory work for #4728

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java [hunk 1]
```diff
@@ -61,19 +65,71 @@ public String getSignalingType() {
...
+                neighborTrack = neighbor;
+                break;
+            }
+        }
+        return neighborTrack;
+    }
+
+    private DiDetector getNextDetector(DiTrackInfra infra, DiTrackEdge curEdge) {
+        // if that fails, look for linked track sections
+        // var curEdge = infra.getEdge(trackSection, Direction.FORWARD);
```
Comment (Khoyo) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java]: That comment should probably be near the call site (what is "that"?)

Also, commented code

File: core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java [hunk 2]
```diff
@@ -61,19 +65,71 @@ public String getSignalingType() {
...
+        while (true) {
+            curEdge = getLinkedTrackSection(infra, curEdge);
+            if (curEdge == null)
+                return null;
```
Comment (Khoyo) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java:89]: Maybe we should warn ?

**Review by Khoyo — APPROVED** [review]
Otherwise lgtm

File: core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java (hunk 2 above)
Comment (multun) [reply core/src/main/java/fr/sncf/osrd/infra/implementation/signaling/modules/bal3/BAL3.java:89]: done right below

**Pull request merged**